src/main/scala/org/ergoplatform/nodeView/ErgoNodeViewHolder.scala

```diff
@@ -386,72 +389,81 @@ abstract class ErgoNodeViewHolder[State <: ErgoState[State]](settings: ErgoSetti
     * which also needs to be propagated to mempool and wallet
     * @param pmod Remote or local persistent modifier
     */
-  protected def pmodModify(pmod: ErgoPersistentModifier, local: Boolean): Unit =
-    if (!history().contains(pmod.id)) {
-      context.system.eventStream.publish(StartingPersistentModifierApplication(pmod))
-
-      log.info(s"Apply modifier ${pmod.encodedId} of type ${pmod.modifierTypeId} to nodeViewHolder")
-
-      history().append(pmod) match {
-        case Success((historyBeforeStUpdate, progressInfo)) =>
-          log.debug(s"Going to apply modifications to the state: $progressInfo")
-          context.system.eventStream.publish(SyntacticallySuccessfulModifier(pmod))
-
-          if (progressInfo.toApply.nonEmpty) {
-            val (newHistory, newStateTry, blocksApplied) =
-              updateState(historyBeforeStUpdate, minimalState(), progressInfo, IndexedSeq.empty)
-
-            newStateTry match {
-              case Success(newMinState) =>
-                val newMemPool = updateMemPool(progressInfo.toRemove, blocksApplied, memoryPool(), newMinState)
-
-                @SuppressWarnings(Array("org.wartremover.warts.OptionPartial"))
-                val v = vault()
-                val newVault = if (progressInfo.chainSwitchingNeeded) {
-                  v.rollback(idToVersion(progressInfo.branchPoint.get)) match {
-                    case Success(nv) => nv
-                    case Failure(e) => log.warn("Wallet rollback failed: ", e); v
+  protected def pmodModify(pmod: ErgoPersistentModifier, local: Boolean): Unit = {
+    if (!history().contains(pmod.id)) { // todo: .contains reads modifier pmod fully here if in db
+
+      // if ADProofs block section generated locally, just dump it into
+      if (pmod.modifierTypeId == ADProofs.modifierTypeId && local) {
+        val bytes = HistoryModifierSerializer.toBytes(pmod) //todo: extra allocation here, eliminate
```

Comment:
Allocation maybe not a problem here (it is not millions per second), but if serialization can be avoided that would be a win.